lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderServiceImpl.java

```diff
@@ -28,190 +31,320 @@
 
   private Context context;
 
-  private boolean mockMode;
-  private LocationEngine locationEngine;
-  private Map<LocationListener, LocationRequest> listenerToRequest;
-  private Map<PendingIntent, LocationRequest> intentToRequest;
-  private Map<LocationCallback, LocationRequest> callbackToRequest;
-  private Map<LocationCallback, Looper> callbackToLooper;
+  private Map<LostApiClient, Map<LocationListener, LocationRequest>> listenerToRequest;
```

Comment:
:D

I like that idea...I've pushed code to expand the `ClientManager` which moves these maps into that class